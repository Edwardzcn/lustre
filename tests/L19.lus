
type r = subrange [-2, 8] of int;

const x : r = -1;

node main(p : int) returns (q : int)
let
  q = x;
tel;

