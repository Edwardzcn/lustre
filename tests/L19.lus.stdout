PhaseToCore
===========

node (p : int)
  returns (q)
  assumes ()
  shows ()
let
  q : int
    = -1
tel

--- Step 1 ---
p : int  = 1
q = -1
--- Step 2 ---
p : int  = 2
q = -1
--- Step 3 ---
p : int  = (EOF)
